### ISSUE TITLE
Contextify installation error

### ISSUE THREAD

**Issue by njwest** [GraphiDocsOrg/docs#25]
Graphidoc still works under these conditions; this is a failure to install an optional dependency, `contextify@0.1.15`, on global graphidocs installation.

OS: macOS 10.14.3
npm v.: 6.4.1 

On `npm install -g @graphidocs/docs`:

```shell
$ npm install -g @graphidocs/docs

/usr/local/bin/graphidocs -> /usr/local/lib/node_modules/@graphidocs/docs/bin/graphidocs.js

> contextify@0.1.15 install /usr/local/lib/node_modules/@graphidocs/docs/node_modules/contextify
> node-gyp rebuild

  CXX(target) Release/obj.target/contextify/src/contextify.o
../src/contextify.cc:131:56: error: no matching member function for call to 'NewInstance'
        Local<Object> wrapper = Nan::New(constructor)->NewInstance();
                                ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~
/Users/username/.node-gyp/10.9.0/include/node/v8.h:3979:44: note: candidate function not viable: requires single argument 'context',
      but no arguments were provided
  V8_WARN_UNUSED_RESULT MaybeLocal<Object> NewInstance(
                                           ^
/Users/username/.node-gyp/10.9.0/include/node/v8.h:3976:44: note: candidate function not viable: requires 3 arguments, but 0 were
      provided
  V8_WARN_UNUSED_RESULT MaybeLocal<Object> NewInstance(
                                           ^
../src/contextify.cc:150:16: error: no member named 'SetAccessCheckCallbacks' in 'v8::ObjectTemplate'
        otmpl->SetAccessCheckCallbacks(GlobalPropertyNamedAccessCheck,
        ~~~~~  ^
../src/contextify.cc:182:80: error: too few arguments to function call, expected 2, have 1
        Local<Value> rv = Nan::New(ctx->sandbox)->GetRealNamedProperty(property);
                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~         ^
/Users/username/.node-gyp/10.9.0/include/node/v8.h:3545:3: note: 'GetRealNamedProperty' declared here
  V8_WARN_UNUSED_RESULT MaybeLocal<Value> GetRealNamedProperty(
  ^
/Users/username/.node-gyp/10.9.0/include/node/v8config.h:416:31: note: expanded from macro 'V8_WARN_UNUSED_RESULT'
#define V8_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
                              ^
../src/contextify.cc:209:67: error: too few arguments to function call, expected 2, have 1
        if (!Nan::New(ctx->sandbox)->GetRealNamedProperty(property).IsEmpty() ||
             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~         ^
/Users/username/.node-gyp/10.9.0/include/node/v8.h:3545:3: note: 'GetRealNamedProperty' declared here
  V8_WARN_UNUSED_RESULT MaybeLocal<Value> GetRealNamedProperty(
  ^
/Users/username/.node-gyp/10.9.0/include/node/v8config.h:416:31: note: expanded from macro 'V8_WARN_UNUSED_RESULT'
#define V8_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
                              ^
../src/contextify.cc:210:71: error: too few arguments to function call, expected 2, have 1
            !Nan::New(ctx->proxyGlobal)->GetRealNamedProperty(property).IsEmpty()) {
             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~         ^
/Users/username/.node-gyp/10.9.0/include/node/v8.h:3545:3: note: 'GetRealNamedProperty' declared here
  V8_WARN_UNUSED_RESULT MaybeLocal<Value> GetRealNamedProperty(
  ^
/Users/username/.node-gyp/10.9.0/include/node/v8config.h:416:31: note: expanded from macro 'V8_WARN_UNUSED_RESULT'
#define V8_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
                              ^
5 errors generated.
make: *** [Release/obj.target/contextify/src/contextify.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:262:23)
gyp ERR! stack     at ChildProcess.emit (events.js:182:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:239:12)
gyp ERR! System Darwin 18.2.0
gyp ERR! command "/usr/local/Cellar/node/10.9.0/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /usr/local/lib/node_modules/@graphidocs/docs/node_modules/contextify
gyp ERR! node -v v10.9.0
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok 
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: contextify@0.1.15 (node_modules/@graphidocs/docs/node_modules/contextify):
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: contextify@0.1.15 install: `node-gyp rebuild`
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: Exit status 1
```

**Comment by mitchellsimoens** [comment]
Thannks for opening an issue. Looking at the dependency [visualization](https://npm.anvaka.com/#/view/2d/%2540graphidocs%252Fdocs), looks like the `contextify` module is coming from `jsdom` which is being required by `docs` which is what is required by this project. Doing a quick search in the code of this project, I'm not seeing where `docs` is being used so will see if removing it doesn't break things and will resolve this issue as a result.

**Comment by mitchellsimoens** [comment]
I should also note in agreement, errors like this are normally safe to ignore.

**Comment by mitchellsimoens** [comment]
Looks like removing `docs` didn't affect anything so I merged the PR. I'll keep this open till the next release is published.

**Comment by mitchellsimoens** [comment]
Just an fyi, I released 1.0.5 today. Sorry for the delay.
